modules/system/classes/UpdateManager.php

```diff
@@ -1046,7 +1046,7 @@ protected function addMessage($class, $message)
      */
     protected function printMessages()
     {
-        if (!count($this->messages)) {
+        if (!count($this->messages) || empty(array_filter(array_flatten($this->messages)))) {
```

Comment:
Is this meant to filter out empty messages? If so we should filter them when they're added instead of on display